rclpy/test/test_executor.py

```diff
@@ -361,16 +379,68 @@ def timer_callback():
         future = Future()
         self.assertFalse(future.done())
         start = time.monotonic()
-        executor.spin_until_future_complete(future=future, timeout_sec=0.1)
+        executor.spin_until_complete(future, timeout_sec=0.1)
         end = time.monotonic()
         # Nothing is ever setting the future, so this should have waited
         # at least 0.1 seconds.
         self.assertGreaterEqual(end - start, 0.1)
         self.assertFalse(future.done())
 
+        start = time.monotonic()
+        executor.spin_until_complete(lambda: False, timeout_sec=0.1)
+        end = time.monotonic()
+        # condition is always false so this should have waited
+        # at least 0.1 seconds.
+        self.assertGreaterEqual(end - start, 0.1)
+        self.assertFalse(future.done())
+
+        timer.cancel()
+
+    def test_executor_spin_until_complete_condition_done(self):
+        self.assertIsNotNone(self.node.handle)
+        executor = SingleThreadedExecutor(context=self.context)
+        executor.add_node(self.node)
+
+        def timer_callback():
+            pass
+        timer = self.node.create_timer(0.003, timer_callback)
```

Comment:
probably same here, how about having timer with `0.1`?

Reply:
Done - plus I've updated other places where a similar pattern of short intervals was used.